jni/include/commons.h

```diff
@@ -48,6 +48,20 @@ namespace knn_jni {
          */
         void freeVectorData(jlong);
 
+        /**
+        * This is utility function that can be used to store signed int8 data in native memory. This function will allocate memory for
+        * the data(rows*columns) with initialCapacity and return the memory address where the data is stored.
+        * If you are using this function for first time use memoryAddress = 0 to ensure that a new memory location is created.
+        * For subsequent calls you can pass the same memoryAddress. If the data cannot be stored in the memory location
+        * will throw Exception.
+        *
+        * @param memoryAddress The address of the memory location where data will be stored.
+        * @param data 2D byte array containing int8 data to be stored in native memory.
+        * @param initialCapacity The initial capacity of the memory location.
+        * @return memory address of std::vector<int8_t> where the data is stored.
+        */
+        jlong storeByteVectorData(knn_jni::JNIUtilInterface *, JNIEnv *, jlong , jobjectArray, jlong);
```

Comment:
should we just have storeByteVectorData rather than having 2. Because at the lowest level both are storing vectors as ByteArray.

Reply:
NVM I saw that binary uses Signed int_8 and byte uses uint_8. I think we might need to move all these storing and freeing up of the memory in their respective IndexService. It can be huge refactor. @heemin32 , @jmazanec15 what is your thought. 

Reason of having all of them in common place is not because of code duplication, I am more worried about ensuring there are not leaks and edge cases are handled during this store and release of memory address.

Reply:
I think we need to have a clear ownership understanding for the data. Someone has to decide when this memory gets freed up. Id probably say its best to keep generic stores here, but have the engines handle cleanup.

Reply:
@jmazanec15 want to understand your comment here. Where are we seeing that ownership of pointers are not correct?

Reply:
Im just saying that we need to ensure the interface makes it clear who has to free what. If we allocate it generically, then free it per engine, this could cause errors